pkg/registries/registries.go

```diff
@@ -131,20 +164,35 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		}
 	}
 
-	// insecureRegistries and blockedRegistries are lists of registries; now that mirrors can be configured at a namespace/repo level,
+	// Add the blocked registry entries to the registries list so that we can find sub-scopes of insecure registries and set both the
+	// blocked and insecure flags accordingly.
+	// e.g *.blocked.insecure.com is a sub-scope of *.insecure.com and should have both the insecure and blocked options set to true. If
+	// we don't add the blocked registries list to the registries config list before going through the insecure registries we won't be able
+	// to check if *.blocked.insecure.com is a sub-scope of *.insecure.com as it won't exist in the registries config list and will not have
+	// insecure=true, so we need to populate the registries config list with the blocked registries before moving on.
+	for _, reg := range blockedRegistries {
+		_ = getRegistryEntry(reg)
+	}
+
+	// insecureRegistries and blockedRegistries are lists of registries or wildcards; now that mirrors can be configured at a namespace/repo level,
 	// configuration at the namespace/repo level would shadow the registry-level entries; so, propagate the insecure/blocked
```

Comment:
```suggestion
	// any of insecureScopes, blockedScopes, and mirrors, can be configured at a namespace/repo level,
	// and in V2RegistriesConf, only the most precise match is used; so, propagate the insecure/blocked
```

---

(BTW this in principle applies to ICSP as well, e.g. mapping `example.com/company` to a mirror + setting `example.com/company/namespace` as insecure would hide the ICSP mapping. I can’t see a good reason for anyone to set insecure/blocked to a sub-scope of an ICSP source, so let’s not worry about it now; if we added `allowedScopes` to be implemented using `registries.conf`, it would be a concern.)

Reply:
Fixed